test/typescript-karma/Dashboards/DataGrid/datagrid.test.js

```diff
@@ -0,0 +1,131 @@
+//@ts-check
+import DataGrid from '/base/code/datagrid/es-modules/masters/datagrid.src.js';
+
+const { test, skip } = QUnit;
+
+//@ts-ignore
+test('DataGrid update methods', async function (assert) {
+    const parentElement = document.getElementById('container');
+    if (!parentElement) {
+        return;
+    }
+
+    const dataTable = new DataGrid.DataTable({
+        columns: {
+            product: ['Apples', 'Pears', 'Plums', 'Bananas'],
+            weight: [100, 40, 0.5, 200],
+            price: [1.5, 2.53, 5, 4.5]
+        }
+    })
+
+    const dataGrid = await DataGrid.dataGrid(parentElement, {
+        dataTable: dataTable,
+        columns: [{
+            id: 'product',
+            headerFormat: 'Column 1',
+            cellFormat: 'before update'
+        }]
+    }, true);
+    dataGrid.viewport?.resizeObserver?.disconnect();
+
+    dataGrid.update({
+        columns: [{
+            id: 'weight',
+            headerFormat: 'Column 2',
+            cellFormat: 'text 1'
+        }, {
+            id: 'product',
+            headerFormat: 'Column 1',
+            cellFormat: 'text 2'
+        }]
+    }, false);
+
+    assert.deepEqual(
+        dataGrid.options.columns[0],
+        {
+            id: 'weight',
+            headerFormat: 'Column 2',
+            cellFormat: 'text 1'
+        },
+        'Update method should rewrite the columns array.'
+    );
+
+    dataGrid.userOptions.columns[0].cellFormat = 'text 3';
+    dataGrid.update(void 0, false);
+
+    assert.strictEqual(
+        dataGrid.options.columns[0].cellFormat,
+        'text 3',
+        'Update method without arguments should update the columns array with the userOptions.'
```

Comment:
Do we want it to work like that? 
The way the options are updated (`dataGrid.userOptions.columns[0].cellFormat = 'text 3';`) is not clear to me...
IMO the `column` should expose the `update` method to update the options.

It might generate some unexpected behaviour 🤔

Reply:
Hmm, but why? For me, this is the clearest option. 😄 That's why we separated `options` from `userOptions`. What do you think @sebastianbochan?

A column can expose an update, but as I described in a comment below, not every ColumnOption has its column instance (e.g. enabled: false).

Reply:
It's clear what is changing when you declare options like this but after that you have to do this strange `update(undefined)` which is counterintuitive for me